homeassistant/components/daikin/switch.py

```diff
@@ -0,0 +1,77 @@
+"""Support for Daikin AirBase zones."""
+import logging
+
+from homeassistant.helpers.entity import ToggleEntity
+
+from . import DOMAIN as DAIKIN_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+ZONE_ICON = 'mdi:home-circle'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Old way of setting up the platform.
+
+    Can only be called when a user accidentally mentions the platform in their
+    config. But even in that case it would have been ignored.
+    """
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Daikin climate based on config_entry."""
+    daikin_api = hass.data[DAIKIN_DOMAIN].get(entry.entry_id)
+    zones = daikin_api.device.zones
+    if zones:
+        async_add_entities([
+            DaikinZoneSwitch(daikin_api, zone_id)
+            for zone_id in range(len(zones))
+        ])
+
+
+class DaikinZoneSwitch(ToggleEntity):
+    """Representation of a zone."""
+
+    def __init__(self, daikin_api, zone_id):
+        """Initialize the zone."""
+        self._api = daikin_api
+        self._zone_id = zone_id
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return "{}-zone{}".format(self._api.mac, self._zone_id)
```

Comment:
Are you sure that zone ID can never change for a zone?

Reply:
Yes it is actually returned from the AC as a comma separated list.